packages/core/src/common/disposable.ts

```diff
@@ -47,6 +47,29 @@ Object.defineProperty(Disposable, 'NULL', {
     }
 });
 
+/**
+ * Utility for tracking a collection of Disposable objects, specifically for
+ * the sake of finding out when all items in the collection have been disposed.
+ *
+ * ```
+ * const dc = new DisposableCollection(myDisposables);
+ * dc.onDispose(() => {
+ *    console.log('All elements in the collection have been disposed');
+ * });
+ * ```
+ *
+ * You can add any number of Disposables to the collection at any time. You
+ * cannot query the contents of the collection. An element is automatically
+ * pruned from the collection when it is disposed.
+ *
+ * The collection will notify only once. It will continue to function
```

Comment:
I'd say this is the main point of this doc besides the general purpose, because I believe that is the most unexpected part of the collections behavior.

Reply:
That's a good point. The collection auto-pruning itself when its elements are disposed is indeed a key value-add, even if you're not making use of the notification feature. OK. Let me revise this and see if we can come up with something we both like :-)

Reply:
@martin-fleck-at I've revamped the description of `DisposableCollection`. I think the new version should address your concerns, and is, frankly, far better thanks to your feedback.

Reply:
Well, thank you for adapting it so quickly and I agree it reads very clear now.